src/card/view/CardView.tsx

```diff
@@ -138,7 +138,7 @@ const NeoCardView = ({
   }, [JSON.stringify(localParameters)]);
 
   useEffect(() => {
-    if (!settingsOpen) {
+    if (!settingsOpen && (selectorChange || type === 'select')) {
```

Comment:
I think it's still confusing here.
Either we need to change the run button to say 'Close' and change the icon if no changes are made (`selectorChange` is false), or, we go back to running every time.

I'm cool with either - as long as it's clear for the user.